File: directionOfReading.cpp
std::vector<int> directionOfReading(std::vector<int> numbers) {
  vector<string> v;
  for (int p : numbers) {
    v.push_back(to_string(p));
  }
  for (string& s : v) {
    while ((int) s.length() < (int) numbers.size()) s = '0' + s;
  }
  vector<int> res;
  int n = numbers.size();
  for (int i = 0; i < n; i++) {
    int cur = 0;
    for (int j = 0; j < n; j++) {
      cur = cur * 10 + (v[j][i] - '0');
    }
    res.push_back(cur);
  }
  return res;
}

File: directionOfReading.js
function directionOfReading(numbers) {
    let strArr = numbers.map(x=>'0'.repeat(numbers.length - x.toString().length) + x)
  let finalnumbers = []
  for(let i=0;i<strArr.length;i++){
    for(let j=0;j<strArr[i].length;j++){
      finalnumbers[j]? finalnumbers[j] += strArr[i][j] : finalnumbers[j] = strArr[i][j]
    }
  }
  return finalnumbers.map(x=>+x)
}
